### DISCUSSION TITLE
Calendar View

### DISCUSSION THREAD

**Discussion by Onatcer** [solidtime-io/solidtime#267] [Feature requests]
Add a view that shows the time entries on a (weekly) calendar view and allows for adding time entries on the calendar.

**Comment by BoThomas** [comment]
Would love this feature!
When using other tools, I really like this to add entries I have missed to enable the timer for or correcting times for past entries.
Feels way more natural to me than using the "Manual time entry" modal.

**Reply by BoThomas** [reply]
I think adding entries should work by simply clicking and dragging on the calendar, bringing up a modal for project input etc. on mouse release; changing time should be a "resize" on the entry.

**Reply by nonameolsson** [reply]
That is how it is working in Toggl. Using that calendar view is my way of tracking  my time.

**Reply by miguel-lhh** [reply]
Indeed! plus, it is much easier to spot overlapping activities, or gaps that you missed when tracking, than in the list view that shows in TimeTracker or Dashboard views.

**Reply by suskozaver** [reply]
+1 for this, also using Toggle, I keep looking for a self hosted solution which would support this... haven't found one yet

**Comment by mathieu-wq** [comment]
The calendar view just came out, and it's amazing! Thanks you so much!

Being the little nitpicker that I am, I have two minor feature requests that could improve its use:

1. Show the current entry: create a block that dynamically grows as time passes. Right now, the current running entry is not shown at all on the calendar view.
2. Add the ability to start a new time entry based on any of the time entries shown in the calendar view (e.g. have a "play" button show up when you hover one of the time entries that, when clicked, starts a new entry with the same project/description and stops a previously running entry)

That being said, thank you for listening to the community and implementing that feature. For some reason, showing the time entries in a calendar view really clicks with my mental model of viewing how I used my time during the past day/week. I'm really excited about using it more.

**Reply by mathieu-wq** [reply]
I might as well throw in a third little request for consideration: having the same "Time Tracker" block at the top of the page to update and start/stop the current running time entry that is seen in the Dashboard and Time views appear on the Calendar view would be the icing on the cake :)

**Comment by korridor** [comment]
This available in solidtime since release [v0.10.0](https://github.com/solidtime-io/solidtime/releases/tag/v0.10.0).
